server/app/services/export/PdfExporter.java

```diff
@@ -84,8 +88,19 @@ private byte[] buildPDF(
             new Paragraph(
                 answerData.questionDefinition().getName(),
                 FontFactory.getFont(FontFactory.HELVETICA_BOLD, 12));
-        Paragraph answer =
-            new Paragraph(answerData.answerText(), FontFactory.getFont(FontFactory.HELVETICA, 11));
+        Paragraph answer = null;
+        if (QuestionType.FILEUPLOAD == answerData.questionDefinition().getQuestionType()) {
+          Path contextPath = answerData.contextualizedPath().join(Scalar.FILE_KEY);
```

Comment:
Can we follow a similar pattern to what we do in JSONExporter ([code](https://github.com/civiform/civiform/blob/14b3f1519fdf2422063d97765212539eda9e3b17/server/app/services/export/JsonExporter.java#L107-L123)) of getting the concrete question type rather than duplicating the extraction code here?

I think it could be something along the lines of:
```
FileUploadQuestion fileUploadQuestion = answerData.applicantQuestion().createFileUploadQuestion();
Optional<String> maybeFileUrl = fileUploadQuestion.getFileKeyValue();
```

I think the current code may imply that we need to handle a missing file (perhaps due to the question being optional)?

Reply:
Very good observation! thank you!